### ISSUE TITLE
Possibility to add attributes to <th> elements

### ISSUE THREAD

**Issue by teslor** [vaadin/vaadin-grid#376]
Example:

```
<thead>
  <tr>
    <th data-id="1st">Built</th>
    <th data-id="2nd">Serial Number</th>
  </tr>
</thead>
```

All attributes are stripped out.

**Comment by Saulis** [comment]
Hi!

To be precise, the element structure defined as a `<table>` under `<vaadin-grid>` isn't actually used as it is, but instead the contents are read and another `<table>` is constructed out of that. When it comes to the `<th>` elements, only `class` and `colspan` attributes are currently read and passed through to the actual data table.

That said, unfortunately there isn't any API to access the header cell elements in a nice way, so you need to fetch using a querySelector.

**Comment by manolo** [comment]
@teslor  Out of curiosity, what is the main reason of having those customised attributes that cannot be covered using custom classnames?

**Comment by teslor** [comment]
@Saulis Thank you for the info. @manolo We would like to have some data-* attributes for header cells. At present we use classes instead & yes it works, however it's not semantically correct.

**Comment by jouni** [comment]
I would like to know more about the use case, why the attributes need to be in the th elements. Is it unacceptable to have them in a contained element inside the th element?

**Comment by jouni** [comment]
Also, we are not fixing this for 1.x, but I’m wondering do we need to fix it for the upcoming 2.0.

**Comment by tomivirkki** [comment]
https://github.com/vaadin/vaadin-grid/issues/1297 should cover this. Closing.
